src/preact.d.ts

```diff
@@ -37,6 +37,16 @@ declare namespace preact {
 		new (props?:PropsType):Component<PropsType, StateType>;
 	}
 
+	interface FunctionalComponent<PropsType> {
+		(props:PropsType & ComponentProps, context?:any):JSX.Element;
+		displayName?:string;
+		defaultProps?:any;
+	}
+
+	// An "inclusive" component is a component that is either a functional
+	// component or a more full-fledged component.
+	type InclusiveComponent<PropsType, StateType> = FunctionalComponent<PropsType> | typeof Component;
```

Comment:
Hmm ... this seems like a useful alias to provide but the name sounds a bit weird. I'm struggling to think of something much better. `AnyComponent`?